op-batcher/batcher/driver.go

```diff
@@ -870,18 +883,22 @@ func (l *BatchSubmitter) recordFailedDARequest(id txID, err error) {
 	if err != nil {
 		l.Log.Warn("DA request failed", logFields(id, err)...)
 	}
-	l.state.TxFailed(id)
+	l.channelMgr.TxFailed(id)
```

Comment:
don't we need to guard this call with the mutex?

Reply:
This is called inside `syncAndPublish`, which already holds the lock.